lib/Backend/BailOut.cpp

```diff
@@ -585,7 +585,7 @@ BailOutRecord::RestoreValues(IR::BailOutKind bailOutKind, Js::JavascriptCallStac
         }
 #endif
 
-        if (scriptContext->IsInDebugMode())
+        if (functionBody->GetUtf8SourceInfo()->IsUtf8SourceInfoInDebugMode())
```

Comment:
I understand that you have added the classname in the Is<classname>InDebugMode to make your refactor easier. But now it does not make sense to have this redundant information. What do you think?